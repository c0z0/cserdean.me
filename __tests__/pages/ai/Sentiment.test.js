import React from "react";
import {
  render,
  cleanup,
  waitForDomChange,
  fireEvent
} from "react-testing-library";
import Router from "next/router";
import "jest-styled-components";

import Sentiment from "../../../pages/ai/sentiment";

const mockedRouter = {
  push: jest.fn(async () => {}),
  prefetch: () => {}
};
Router.router = mockedRouter;

afterEach(cleanup);

test("Sentiment page matches snapshot", async () => {
  global.fetch = jest.fn(async () => ({ ok: true }));

  const { container } = render(<Sentiment />);

  expect(container.firstChild).toMatchSnapshot();

  await waitForDomChange({ container });

  expect(container.firstChild).toMatchSnapshot();
});

test("Sentiment page shows loading message and then input", async () => {
  global.fetch = jest.fn(async () => ({ ok: true }));

  const { container, queryByTestId, queryByText } = render(<Sentiment />);

  expect(queryByText("Loading...")).toBeTruthy();

  await waitForDomChange({ container });

  expect(queryByTestId("input-element")).toBeTruthy();
});

test("Sentiment page shows loading message and then error", async () => {
  global.fetch = jest.fn(async () => ({ ok: false }));

  const { container, queryByText } = render(<Sentiment />);

  expect(queryByText("Loading...")).toBeTruthy();

  await waitForDomChange({ container });

  expect(queryByText("Something went wrong")).toBeTruthy();
});

test("Sentiment page shows loading message and then catches error", async () => {
  global.fetch = jest.fn(async () => {
    throw new Error("test-error");
  });

  const { container, queryByText } = render(<Sentiment />);

  expect(queryByText("Loading...")).toBeTruthy();

  await waitForDomChange({ container });

  expect(queryByText("Something went wrong")).toBeTruthy();
});

test("Predict not ok", async () => {
  global.fetch = jest.fn(async () => ({ ok: true }));

  const { container, queryByTestId, queryByText } = render(<Sentiment />);

  await waitForDomChange({ container });

  const input = queryByTestId("input-element");

  global.fetch = jest.fn(async () => ({
    ok: false
  }));

  fireEvent.change(input, {
    target: { value: "test-value" }
  });

  await waitForDomChange({ container });

  expect(queryByText("Something went wrong")).toBeTruthy();
});

test("Predict not ok", async () => {
  global.fetch = jest.fn(async () => ({ ok: true }));

  const { container, queryByTestId, queryByText } = render(<Sentiment />);

  await waitForDomChange({ container });

  const input = queryByTestId("input-element");

  global.fetch = jest.fn(async () => {
    throw new Error("test-error");
  });

  fireEvent.change(input, {
    target: { value: "test-value" }
  });

  await waitForDomChange({ container });

  expect(queryByText("Something went wrong")).toBeTruthy();
});

test("Sentiment page predicts and displays and clears input", async () => {
  global.fetch = jest.fn(async () => ({ ok: true }));

  const { container, queryByText, queryByTestId, getByTestId } = render(
    <Sentiment />
  );

  expect(queryByText("Loading...")).toBeTruthy();

  await waitForDomChange({ container });

  const input = queryByTestId("input-element");

  global.fetch = jest.fn(async () => ({
    ok: true,
    json: async () => ({
      prediction: 4
    })
  }));

  fireEvent.change(input, {
    target: { value: "test-value" }
  });

  expect(input.value).toBe("test-value");
  expect(queryByTestId("input-loading")).toBeTruthy();

  await waitForDomChange({ container });

  expect(queryByTestId("input-loading")).toBeNull();

  expect(container.querySelector("span")).toHaveStyleRule("left", "300px");

  fireEvent.click(getByTestId("input-clear-button"));

  expect(input.value).toBe("");
  expect(container.querySelector("span")).toHaveStyleRule("left", "150px");

  fireEvent.change(input, {
    target: { value: "test-value" }
  });

  await waitForDomChange({ container });

  fireEvent.change(input, {
    target: { value: "" }
  });

  expect(container.querySelector("span")).toHaveStyleRule("left", "150px");
});
