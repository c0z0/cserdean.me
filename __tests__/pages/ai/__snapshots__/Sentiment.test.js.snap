// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sentiment page matches snapshot 1`] = `
.c1 {
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI','Roboto','Oxygen', 'Ubuntu','Cantarell','Fira Sans',serif;
  background: none;
  padding-top: 20px;
  position: static;
  z-index: 100;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 20px;
  margin-top: 0;
}

.c3 {
  height: 40px;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  margin-right: 20px;
  -webkit-transition: all 0.5s ease-in-out;
  transition: all 0.5s ease-in-out;
  box-shadow: 0 0 0 0 rgba(0,0,0,0);
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.c4 {
  background: none;
  border: none;
  outline: none;
  box-shadow: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  width: 40px;
  height: 40px;
  padding: 0;
}

.c5 * {
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  width: 22px;
  height: 1px;
  background-color: white;
}

.c6 {
  -webkit-transform: rotate(0) translateY(-4px);
  -ms-transform: rotate(0) translateY(-4px);
  transform: rotate(0) translateY(-4px);
}

.c7 {
  -webkit-transform: rotate(0) translateY(4px);
  -ms-transform: rotate(0) translateY(4px);
  transform: rotate(0) translateY(4px);
}

.c8 {
  padding-top: 20px;
  display: none;
}

.c9 {
  color: white;
  font-size: 12px;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: block;
  padding: 16px;
  padding-left: 20px;
  border-bottom: 1px solid rgba(255,255,255,.2);
  text-transform: uppercase;
}

.c10 {
  color: white;
  font-size: 12px;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: block;
  padding: 16px;
  padding-left: 20px;
  border-bottom: 1px solid rgba(255,255,255,.2);
  text-transform: uppercase;
  background: rgba(255,255,255,.2);
  border-color: rgba(0,0,0,0);
}

.c11 {
  color: white;
  font-size: 12px;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: block;
  padding: 16px;
  padding-left: 20px;
  border-bottom: 1px solid rgba(255,255,255,.2);
  text-transform: uppercase;
  color: #FE51BB;
}

.c0 {
  font-size: 12px;
  color: white;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI','Roboto','Oxygen', 'Ubuntu','Cantarell','Fira Sans',serif;
  max-width: 900px;
  margin: 0 auto;
}

.c12 {
  margin: 80px 0;
}

.c13 {
  text-align: center;
}

.c14 {
  font-weight: normal;
  font-size: 1.5em;
}

.c15 {
  color: #3997F9;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: Menlo,Monaco,Lucida Console,Liberation Mono,DejaVu Sans Mono, Bitstream Vera Sans Mono,Courier New,monospace,serif;
}

.c16 {
  opacity: 0.2;
  font-size: 16px;
}

.c17 {
  margin: 0 auto;
  position: relative;
  height: 2px;
  width: 300px;
  background: linear-gradient(to right,#E52D2D,#C928A3,#288BC9,#28CCB3,#28CC6C);
}

.c18 {
  position: absolute;
  top: -5px;
  bottom: -5px;
  background: white;
  width: 2px;
  left: 150px;
  -webkit-transition: all 0.5s;
  transition: all 0.5s;
}

@media (min-width:992px) {
  .c1 {
    width: 100%;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    background: none;
  }
}

@media (min-width:992px) {
  .c2 {
    margin: 0;
  }
}

@media (min-width:992px) {
  .c4 {
    display: none;
  }
}

@media (min-width:992px) {
  .c8 {
    padding: 0;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    font-size: 12px;
    -webkit-flex: 1;
    -ms-flex: 1;
    flex: 1;
  }
}

@media (min-width:992px) {
  .c9 {
    margin: 0 10px;
    display: inline-block;
    padding: 0;
    color: rgba(255,255,255,0.66);
    text-align: left;
    border-bottom: none;
    text-transform: none;
    -webkit-transition: all 0.2s linear;
    transition: all 0.2s linear;
  }

  .c9:hover {
    color: white;
  }
}

@media (min-width:992px) {
  .c10 {
    margin: 0 10px;
    display: inline-block;
    padding: 0;
    color: rgba(255,255,255,0.66);
    text-align: left;
    border-bottom: none;
    text-transform: none;
    -webkit-transition: all 0.2s linear;
    transition: all 0.2s linear;
    color: white;
    background: none;
  }

  .c10:hover {
    color: white;
  }
}

@media (min-width:992px) {
  .c11 {
    margin: 0 10px;
    display: inline-block;
    padding: 0;
    color: rgba(255,255,255,0.66);
    text-align: left;
    border-bottom: none;
    text-transform: none;
    -webkit-transition: all 0.2s linear;
    transition: all 0.2s linear;
    border-radius: 2rem;
    background-color: #FE51BB;
    padding: 0.625rem 1.5rem;
    color: white;
    margin: 0;
    margin-left: 20px;
  }

  .c11:hover {
    color: white;
  }
}

@media (max-width:992px) {
  .c12 {
    margin: 40px 20px;
  }
}

@media (min-width:992px) {
  .c13 {
    position: absolute;
    top: 50%;
    left: 50%;
    -webkit-transform: translate(-50%,-50%);
    -ms-transform: translate(-50%,-50%);
    transform: translate(-50%,-50%);
    left: 50%;
  }
}

<div
  class="c0"
>
  <header>
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <a
          data-testid="menu-logo"
          href="/"
        >
          <img
            alt="logo"
            class="c3"
            src="data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNDE2IiBoZWlnaHQ9IjEyMjAiIHZpZXdCb3g9IjAgMCA3MDggNjEwIj4gIDxkZWZzPiAgICA8c3R5bGU+ICAgICAgLmNscy0xIHsgICAgICAgIGZpbGw6ICNDMUMxQzE7ICAgICAgfSAgICAgIC5jbHMtMSwgLmNscy0yLCAuY2xzLTMgeyAgICAgICAgZmlsbC1ydWxlOiBldmVub2RkOyAgICAgIH0gICAgICAuY2xzLTIgeyAgICAgICAgZmlsbDogI0QwRDBEMDsgICAgICB9ICAgICAgLmNscy0zIHsgICAgICAgIGZpbGw6ICNCNkI2QjY7ICAgICAgfSAgICA8L3N0eWxlPiAgPC9kZWZzPiAgPHBhdGggY2xhc3M9ImNscy0xIiBkPSJNNTg4Ljc4NiwyNTguOTY0bDUzLjQ0MSw5NS4yMDdMNDkxLjMzNCw2MTMuMzQySDk5NC4zMUw5NDQuMDEzLDcwNUgzNDAuNDQxWiIgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoLTI4NyAtOTUpIi8+ICA8cGF0aCBjbGFzcz0iY2xzLTIiIGQ9Ik0yODcsNjEzLjM0MkwzNDAuNDQxLDcwNSw1ODguNzg2LDI2NS42NzNsMTUwLjg5MywyNjQuOGgxMDAuNkw1OTEuOTI5LDk1WiIgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoLTI4NyAtOTUpIi8+ICA8cGF0aCBjbGFzcz0iY2xzLTMiIGQ9Ik01MzguNjQsNTMxLjE2Nkw0OTAuOCw2MTMuMzQySDk5NUw2OTguMTIyLDk1LjU0SDU5MS4yNEw4MzguNTA5LDUyOC4wMDVaIiB0cmFuc2Zvcm09InRyYW5zbGF0ZSgtMjg3IC05NSkiLz48L3N2Zz4="
            title="source"
          />
        </a>
        <button
          class="c4"
          data-testid="menu-button"
          type="button"
        >
          <div
            class="c5"
          >
            <div
              class="c6"
            />
            <div
              class="c7"
            />
          </div>
        </button>
      </div>
      <div
        class="c8"
        data-testid="menu-wrapper"
      >
        <div>
          <a
            class="c9"
            data-testid="menu-item-target-/"
            href="/"
          >
            Home
          </a>
          <a
            class="c9"
            data-testid="menu-item-target-/projects"
            href="/projects"
          >
            Projects
          </a>
          <a
            class="c10"
            data-testid="menu-item-target-/ai"
            href="/ai"
          >
            AI Experiments
          </a>
        </div>
        <div>
          <a
            class="c9"
            data-testid="menu-item-target-/static/cv_cosmin_serdean.pdf"
            href="/static/cv_cosmin_serdean.pdf"
            rel="noopener noreferrer"
            target="_blank"
          >
            CV
          </a>
          <a
            class="c9"
            data-testid="menu-item-target-https://github.com/c0z0"
            href="https://github.com/c0z0"
            rel="noopener noreferrer"
            target="_blank"
          >
            GitHub
          </a>
          <a
            class="c11"
            data-testid="menu-item-target-mailto:cosmoserdean@gmail.com"
            href="mailto:cosmoserdean@gmail.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            Email
          </a>
        </div>
      </div>
    </div>
  </header>
  <div
    class="c12"
  >
    <div
      class="c13"
    >
      <h4
        class="c14"
      >
        Sentiment classifier trained on movie reviews
         
        <a
          class="c15"
          href="https://github.com/c0z0/sentiment-classifier"
        >
          [src]
        </a>
      </h4>
      <p
        class="c16"
      >
        Loading...
      </p>
      <div
        class="c17"
      >
        <span
          class="c18"
        />
      </div>
    </div>
  </div>
  <style>
    
        body {
          background: black;
        }
      
  </style>
</div>
`;

exports[`Sentiment page matches snapshot 2`] = `
<div
  class="sc-dnqmqq cRwGLD"
>
  <header>
    <div
      class="sc-bdVaJa cvLuQB"
    >
      <div
        class="sc-bwzfXH bsmKCX"
      >
        <a
          data-testid="menu-logo"
          href="/"
        >
          <img
            alt="logo"
            class="sc-htpNat cqNGZM"
            src="data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNDE2IiBoZWlnaHQ9IjEyMjAiIHZpZXdCb3g9IjAgMCA3MDggNjEwIj4gIDxkZWZzPiAgICA8c3R5bGU+ICAgICAgLmNscy0xIHsgICAgICAgIGZpbGw6ICNDMUMxQzE7ICAgICAgfSAgICAgIC5jbHMtMSwgLmNscy0yLCAuY2xzLTMgeyAgICAgICAgZmlsbC1ydWxlOiBldmVub2RkOyAgICAgIH0gICAgICAuY2xzLTIgeyAgICAgICAgZmlsbDogI0QwRDBEMDsgICAgICB9ICAgICAgLmNscy0zIHsgICAgICAgIGZpbGw6ICNCNkI2QjY7ICAgICAgfSAgICA8L3N0eWxlPiAgPC9kZWZzPiAgPHBhdGggY2xhc3M9ImNscy0xIiBkPSJNNTg4Ljc4NiwyNTguOTY0bDUzLjQ0MSw5NS4yMDdMNDkxLjMzNCw2MTMuMzQySDk5NC4zMUw5NDQuMDEzLDcwNUgzNDAuNDQxWiIgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoLTI4NyAtOTUpIi8+ICA8cGF0aCBjbGFzcz0iY2xzLTIiIGQ9Ik0yODcsNjEzLjM0MkwzNDAuNDQxLDcwNSw1ODguNzg2LDI2NS42NzNsMTUwLjg5MywyNjQuOGgxMDAuNkw1OTEuOTI5LDk1WiIgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoLTI4NyAtOTUpIi8+ICA8cGF0aCBjbGFzcz0iY2xzLTMiIGQ9Ik01MzguNjQsNTMxLjE2Nkw0OTAuOCw2MTMuMzQySDk5NUw2OTguMTIyLDk1LjU0SDU5MS4yNEw4MzguNTA5LDUyOC4wMDVaIiB0cmFuc2Zvcm09InRyYW5zbGF0ZSgtMjg3IC05NSkiLz48L3N2Zz4="
            title="source"
          />
        </a>
        <button
          class="sc-bxivhb cGjZfZ"
          data-testid="menu-button"
          type="button"
        >
          <div
            class="sc-ifAKCX bTSODv"
          >
            <div
              class="sc-EHOje gxQiPh"
            />
            <div
              class="sc-bZQynM hrtvGh"
            />
          </div>
        </button>
      </div>
      <div
        class="sc-gzVnrw jmEgEA"
        data-testid="menu-wrapper"
      >
        <div>
          <a
            class="sc-htoDjs elorzL"
            data-testid="menu-item-target-/"
            href="/"
          >
            Home
          </a>
          <a
            class="sc-htoDjs elorzL"
            data-testid="menu-item-target-/projects"
            href="/projects"
          >
            Projects
          </a>
          <a
            class="sc-htoDjs hKRjuU"
            data-testid="menu-item-target-/ai"
            href="/ai"
          >
            AI Experiments
          </a>
        </div>
        <div>
          <a
            class="sc-htoDjs elorzL"
            data-testid="menu-item-target-/static/cv_cosmin_serdean.pdf"
            href="/static/cv_cosmin_serdean.pdf"
            rel="noopener noreferrer"
            target="_blank"
          >
            CV
          </a>
          <a
            class="sc-htoDjs elorzL"
            data-testid="menu-item-target-https://github.com/c0z0"
            href="https://github.com/c0z0"
            rel="noopener noreferrer"
            target="_blank"
          >
            GitHub
          </a>
          <a
            class="sc-htoDjs zizbk"
            data-testid="menu-item-target-mailto:cosmoserdean@gmail.com"
            href="mailto:cosmoserdean@gmail.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            Email
          </a>
        </div>
      </div>
    </div>
  </header>
  <div
    class="sc-iwsKbI ceQLln"
  >
    <div
      class="sc-fjdhpX bisAVX"
    >
      <h4
        class="sc-jzJRlG lPSqz"
      >
        Sentiment classifier trained on movie reviews
         
        <a
          class="sc-cSHVUG leHIdH"
          href="https://github.com/c0z0/sentiment-classifier"
        >
          [src]
        </a>
      </h4>
      .c0 {
  position: relative;
  display: inline-block;
  margin-top: 16px;
  margin-bottom: 32px;
}

.c1 {
  width: 290px;
  height: 18px;
  border: none;
  border-bottom: solid rgba(255,255,255,0.4) 1px;
  padding: 8px 0;
  padding-right: 10px;
  outline: none;
  font-size: 16px;
  background: none;
  -webkit-transition: 0.2s all;
  transition: 0.2s all;
  color: white;
}

.c1:focus {
  border-color: rgba( 255, 255, 255, 1 );
}

<div
        class="c0"
      >
        <input
          class="c1"
          data-testid="input-element"
          placeholder="Type something..."
          type="text"
          value=""
        />
      </div>
      <div
        class="sc-kgoBCf lfNGqQ"
      >
        <span
          class="sc-kGXeez dmbOrh"
        />
      </div>
    </div>
  </div>
  <style>
    
        body {
          background: black;
        }
      
  </style>
</div>
`;
